pkg/pillar/scripts/get-alpine-pkg-source.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+# Either takes one or more -t <tag>, or a single EVE version
+# e.g.,
+#  get-alpine-pkg-source.sh -t lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64
+# or
+#  get-alpine-pkg-source.sh 0.0.0-master-9755360f-kvm-amd64
+
+verbose=
+tags=
+while getopts vt: o
+do      case "$o" in
+        v)      verbose=1;;
+        t)      tags="$tags $OPTARG";;
+        [?])    echo "Usage1: $0 [-v] [-t <tag>]+ [<version>]"
+                exit 1;;
+        esac
+done
+shift $((OPTIND-1))
+
+if [ $# == 0 ] && [ -z "$tags" ]; then
+    echo "Usage2: $0 [-v] [-t <tag>]+ [<version>]"
+    exit 1
+fi
+if [ $# -gt 1 ]; then
+    echo "Usage3: $0 [-v] [-t <tag>]+ [<version>]"
+    exit 1
+fi
+
+if [ $# == 1 ]; then
+    VERSION=$1
+    [ -n "$verbose" ] && echo "retrieving build_config for $VERSION"
+    docker run "lfedge/eve:${VERSION}" build_config >build_config.out
+    tags=$(grep image: build_config.out | awk '{print $2}' | sed 's,^docker.io/,,')
+fi
+
+# Create tmpdir and cd there?
+outdir=/tmp/$$
+mkdir -p $outdir
+cd $outdir || exit
+
+[ -n "$verbose" ] && echo "outdir: $outdir"
+
+INSTALLED=/tmp/installed.$$
+OCPAIRS=/tmp/ocpairs.$$
+truncate --size 0 ${OCPAIRS}
+
+for TAG in ${tags}; do
+    [ -n "$verbose" ] && echo "processing $TAG"
+    rm -f ${INSTALLED}
+    c=$(docker create "${TAG}")
+    docker cp "$c":/lib/apk/db/installed ${INSTALLED}
+    docker rm "$c"
+
+    if [ ! -f ${INSTALLED} ]; then
+	echo "No /lib/apk/db/installed for ${TAG}"
+	continue
+    fi
+
+    # Get pairs of origin and commit. Assumes commit is after origin
+    # otherwise we produce XXX output string
+    # Filter out duplicates
+    awk -F: '
+        /^o:/ { origin=$2 } 
+	/^c:/ { commit=$2; print origin, commit; origin="XXX" }' < ${INSTALLED} | sort -u >> ${OCPAIRS}
+    # shellcheck disable=SC2002
+    [ -n "$verbose" ] && echo "after processing $TAG: $(cat ${OCPAIRS} | wc -l) packages"
+done
+
+# shellcheck disable=SC2002
+[ -n "$verbose" ] && echo "found $(cat ${OCPAIRS} |wc -l) packages $(sort -u ${OCPAIRS} | wc -l) unique"
+
+# shellcheck disable=SC2002
+cat ${OCPAIRS} | sort -u | while read -r line ; do
+    # shellcheck disable=SC2086
+    set -- $line
+    origin=$1
+    commit=$2
+    # Include commit in directory to handle different versions
+    dstdir="${origin}.${commit}"
+    [ -n "$verbose" ] && echo "origin: ${origin} commit: ${commit} dstdir: ${dstdir}"
+    pkgurl="https://git.alpinelinux.org/aports/plain/main/${origin}/APKBUILD?id=${commit}"
```

Comment:
I have to check the logs, but I think I managed to fetch all of the alpine stuff using main.

Reply:
Can you see the source files for [qemu-img](https://github.com/lf-edge/eve/blob/fbb85e40f957b43f3d047a43250f213b2b50e68f/pkg/pillar/Dockerfile#L23) which is [located](https://github.com/lf-edge/eve/blob/fbb85e40f957b43f3d047a43250f213b2b50e68f/pkg/alpine/mirrors/3.16/community#L4) inside community?

_FWIW I can see something like `fts.1eb911922a7e818a6f314a68c0561bba008e6efd/APKBUILD: line 1: Too: command not found` and re-download of previous sourcing results (i.e. busybox download again and again) because curl stores `Too Many Requestss` inside `APKBUILD` instead of real content. But script still continues to download without explicit error._

Reply:
Fixed